File: Dia10/d10p1/src/main.js
readFileSync = require('fs').readFileSync;

let data = readFileSync('../../input.txt', 'utf8').split('\n')//.map((v, i, arr) => parseInt(v.trim()));
let count = 0;
for(let line of data) {
	stack = []
	// console.log('"'+line+'"')
	for (let c of line.trim()) {
		if (c !== undefined) {
			if ('([{<'.includes(c)) {
				stack.push(c)
			} else {
				x = stack.pop()
				if ((x == '(' && c != ')') || (x != '(' && c.charCodeAt(0) - x.charCodeAt(0) != 2)) {
					if (c == ')') {
						count += 3
					} else if (c == ']') {
						count += 57
					} else if (c == '}') {
						count += 1197
					} else if (c == '>') {
						count += 25137
					} break
				}
			}
		}
		
	}
}

console.log(count);

File: Dia10/d10p1/src/Main.java
import java.io.FileInputStream;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		try {
			// the file to be opened for reading
			FileInputStream fis = new FileInputStream("../../input.txt");
			Scanner sc = new Scanner(fis); // file to be scanned
			ArrayList<String> nums = new ArrayList<String>();
			// returns true if there is another line to read
			while (sc.hasNextLine()) {
				nums.add(sc.nextLine()); // returns the line that was skipped
			}
			int count = 0;
			for (int i = 0; i < nums.size(); i++) {
				String s = nums.get(i).trim();
				char[] stack = new char[s.length()];
				int size = 0;
				for (int j = 0; j < s.length(); j++) {
					char c = s.charAt(j);
					if ("([{<".indexOf(c) != -1) {
						stack[size++] = c;
					} else {
						char x = stack[--size];
						if ((x == '(' && c != ')') || (x != '(' && c - x != 2)) {
							if (c == ')') {
								count += 3;
							} else if (c == ']') {
								count += 57;
							} else if (c == '}') {
								count += 1197;
							} else if (c == '>') {
								count += 25137;
							}
							break;
						}
					}

				}
			}
			System.out.println(count);
			sc.close(); // closes the scanner
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}
